src/sdk/namespace_fs.js

```diff
@@ -1777,16 +1780,66 @@ class NamespaceFS {
     ////////////////////
 
     async get_object_tagging(params, object_sdk) {
-        // TODO
-        return { tagging: [] };
+        const tag_set = [];
+        const file_path = this._get_file_path(params);
+        try {
+            const fs_context = this.prepare_fs_context(object_sdk);
+            dbg.log0('namespace_fs.get_object_tagging: param ', params, 'file_path :', file_path);
+            const file = await nb_native().fs.open(fs_context, file_path);
+            const stat = await file.stat(fs_context);
+            if (stat.xattr) {
+                for (const [xattr_key, xattr_value] of Object.entries(stat.xattr)) {
+                    if (xattr_key.includes(XATTR_NOOBAA_CUSTOM_PREFIX)) {
+                        tag_set.push({
+                            key: xattr_key.replace(XATTR_NOOBAA_CUSTOM_PREFIX, ''),
+                            value: xattr_value,
+                        });
+                    }
+                }
+            }
+        } catch (err) {
+            dbg.error(`namespace_fs.get_object_tagging: failed in dir ${file_path} with error: `, err);
+            throw this._translate_object_error_codes(err);
+        }
+        dbg.log0('namespace_fs.get_object_tagging: return tagging ', tag_set, 'file_path :', file_path);
+        return { tagging: tag_set };
     }
+
     async delete_object_tagging(params, object_sdk) {
-        // TODO
-        return {};
+        dbg.log0('namespace_fs.delete_object_tagging:', params);
+        const file_path = this._get_file_path(params);
+        const fs_context = this.prepare_fs_context(object_sdk);
+        try {
+            await this._clear_user_xattr(fs_context, file_path, XATTR_NOOBAA_CUSTOM_PREFIX);
+        } catch (err) {
+            dbg.error(`namespace_fs.delete_object_tagging: failed in dir ${file_path} with error: `, err);
+            throw this._translate_object_error_codes(err);
+        }
+        //ideal case put should return version id, but nsfs do not support versioning. adding dummy version.
+        return {version_id: 'v1'};
     }
+
     async put_object_tagging(params, object_sdk) {
-        // TODO
-        return { tagging: [] };
+        let fs_xattr;
+        const tagging = params.tagging && Object.fromEntries(params.tagging.map(tag => ([tag.key, tag.value])));
+        for (const [xattr_key, xattr_value] of Object.entries(tagging)) {
+            fs_xattr = Object.assign(fs_xattr || {}, {
+                [XATTR_NOOBAA_CUSTOM_PREFIX + xattr_key]: xattr_value,
+            });
+        }
```

Comment:
I think we can avoid Object.assign by initiating the fs_xattr as an object:
```suggestion
        let fs_xattr = {};
        const tagging = params.tagging && Object.fromEntries(params.tagging.map(tag => ([tag.key, tag.value])));
        for (const [xattr_key, xattr_value] of Object.entries(tagging)) {
              fs_xattr[XATTR_NOOBAA_CUSTOM_PREFIX + xattr_key] = xattr_value;
        }
```
Am I missing something?